src/metrics/include/metrics/metrics.hpp

```diff
@@ -473,9 +473,18 @@ class Timer {
 class OperationContext {
 
 public:
-    OperationContext(v1::OperationImpl& op)
+    explicit OperationContext(v1::OperationImpl& op)
         : _op{std::addressof(op)}, _started{metrics::clock::now()} {}
 
+    OperationContext(OperationContext&& rhs) noexcept
+        : _op{std::move(rhs._op)},
+          _started{std::move(rhs._started)},
+          _isClosed{std::move(rhs._isClosed)},
```

Comment:
[optional] Consider writing `_isClosed{std::exchange(rhs._isClosed, true)}` and removing line 485.

Reply:
Interesting. Considered but that seemed harder to reason about so left as-is 😄